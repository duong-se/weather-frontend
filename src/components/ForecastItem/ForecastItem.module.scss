.root{
  width: 7rem;
  height: 7rem;
  font-size: 0.8rem;
  .weatherStateImg {
    width: 1rem;
    margin: 0 auto;
  }
  .textFontSize {
    font-size: 0.8rem;
  }
  .temperature {
    justify-content: space-evenly;
  }
}