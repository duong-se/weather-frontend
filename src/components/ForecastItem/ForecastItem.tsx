import React from 'react'
import { ConsolidatedWeather } from '../../typings'
import { roundNumber } from '../../utils/roundNumber'
import { getDay } from '../../utils/getDay'
import style from './ForecastItem.module.scss'

export type ForeCastItemProps = {
  consolidatedWeather: ConsolidatedWeather
  onClick: (e: React.MouseEvent<HTMLButtonElement>) => void
}

export const ForecastItem: React.FC<ForeCastItemProps> = ({ consolidatedWeather, onClick }) => {
  return (
    <button className={`m-1 btn btn-light ${style.root}`} onClick={onClick}>
      <h6 className={'card-subtitle mb-2 mt-2 text-muted'}>{getDay(consolidatedWeather?.applicable_date)}</h6>
      <div className={`${style.weatherStateImg} mb-2`}>
        <img
          alt={consolidatedWeather?.weather_state_abbr}
          src={`/static/img/weather/${consolidatedWeather?.weather_state_abbr}.svg`}
        />
      </div>
      <div className={`row card-text d-flex ${style.temperature}`}>
        <h6 data-test="min-temp">{roundNumber(consolidatedWeather?.min_temp)}&#8451;</h6>
        <h6 data-test="max-temp">{roundNumber(consolidatedWeather?.max_temp)}&#8451;</h6>
      </div>
    </button>
  )
}

export default ForecastItem
