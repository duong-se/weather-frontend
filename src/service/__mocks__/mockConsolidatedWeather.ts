export const mockConsolidatedWeather = {
  "id": 6547671268458496,
  "weather_state_name": "Light Cloud",
  "weather_state_abbr": "lc",
  "wind_direction_compass": "SW",
  "created": "2020-09-22T15:20:01.480287Z",
  "applicable_date": "2020-09-22",
  "min_temp": 14.78,
  "max_temp": 24,
  "the_temp": 24.035,
  "wind_speed": 5.706272015681752,
  "wind_direction": 217.66164574425014,
  "air_pressure": 1008.5,
  "humidity": 51,
  "visibility": 9.400414081762506,
  "predictability": 70
}
