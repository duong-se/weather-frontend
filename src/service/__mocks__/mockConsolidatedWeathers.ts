export const mockConsolidatedWeathers = [
  {
    "id": 6547671268458496,
    "weather_state_name": "Light Cloud",
    "weather_state_abbr": "lc",
    "wind_direction_compass": "SW",
    "created": "2020-09-22T15:20:01.480287Z",
    "applicable_date": "2020-09-22",
    "min_temp": 14.78,
    "max_temp": 24,
    "the_temp": 24.035,
    "wind_speed": 5.706272015681752,
    "wind_direction": 217.66164574425014,
    "air_pressure": 1008.5,
    "humidity": 51,
    "visibility": 9.400414081762506,
    "predictability": 70
  },
  {
    "id": 4713030018400256,
    "weather_state_name": "Heavy Rain",
    "weather_state_abbr": "hr",
    "wind_direction_compass": "SW",
    "created": "2020-09-22T15:20:02.471805Z",
    "applicable_date": "2020-09-23",
    "min_temp": 11.469999999999999,
    "max_temp": 19.13,
    "the_temp": 18.705,
    "wind_speed": 7.6319694457806415,
    "wind_direction": 229.33431218602843,
    "air_pressure": 1001.5,
    "humidity": 70,
    "visibility": 8.366141732283463,
    "predictability": 77
  },
  {
    "id": 6312566033743872,
    "weather_state_name": "Heavy Rain",
    "weather_state_abbr": "hr",
    "wind_direction_compass": "SW",
    "created": "2020-09-22T15:20:02.052900Z",
    "applicable_date": "2020-09-24",
    "min_temp": 8.844999999999999,
    "max_temp": 15.34,
    "the_temp": 15.274999999999999,
    "wind_speed": 9.736295492422917,
    "wind_direction": 225.3331262999033,
    "air_pressure": 996.5,
    "humidity": 59,
    "visibility": 14.177205122087011,
    "predictability": 77
  },
  {
    "id": 4546612517928960,
    "weather_state_name": "Light Rain",
    "weather_state_abbr": "lr",
    "wind_direction_compass": "NNW",
    "created": "2020-09-22T15:20:01.658749Z",
    "applicable_date": "2020-09-25",
    "min_temp": 7.495,
    "max_temp": 13.82,
    "the_temp": 13.04,
    "wind_speed": 10.535621595103263,
    "wind_direction": 334.5010293182893,
    "air_pressure": 1004.5,
    "humidity": 61,
    "visibility": 14.033979062276305,
    "predictability": 75
  },
  {
    "id": 4604599405117440,
    "weather_state_name": "Clear",
    "weather_state_abbr": "c",
    "wind_direction_compass": "NNW",
    "created": "2020-09-22T15:20:02.569048Z",
    "applicable_date": "2020-09-26",
    "min_temp": 7.385,
    "max_temp": 13.649999999999999,
    "the_temp": 13.485,
    "wind_speed": 8.837651108873512,
    "wind_direction": 330.16248530421626,
    "air_pressure": 1015,
    "humidity": 48,
    "visibility": 17.28157559850473,
    "predictability": 68
  },
  {
    "id": 6338147862446080,
    "weather_state_name": "Clear",
    "weather_state_abbr": "c",
    "wind_direction_compass": "NW",
    "created": "2020-09-22T15:20:04.446373Z",
    "applicable_date": "2020-09-27",
    "min_temp": 5.655,
    "max_temp": 14.48,
    "the_temp": 12.6,
    "wind_speed": 4.190453948938201,
    "wind_direction": 314.5,
    "air_pressure": 1021,
    "humidity": 47,
    "visibility": 9.999726596675416,
    "predictability": 68
  }
]