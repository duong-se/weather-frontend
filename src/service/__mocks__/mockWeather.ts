export const mockWeather = {
  "consolidated_weather": [
    {
      "air_pressure": 1009.5,
      "applicable_date": "2020-09-23",
      "created": "2020-09-23T16:00:46.374603Z",
      "humidity": 71,
      "id": 5728148488454144,
      "max_temp": 33.65,
      "min_temp": 25.015,
      "predictability": 77,
      "the_temp": 31.549999999999997,
      "visibility": 8.473638948540524,
      "weather_state_abbr": "hr",
      "weather_state_name": "Heavy Rain",
      "wind_direction": 157.83272289145478,
      "wind_direction_compass": "SSE",
      "wind_speed": 4.518443377942151
    },
    {
      "air_pressure": 1011,
      "applicable_date": "2020-09-24",
      "created": "2020-09-23T16:00:49.298547Z",
      "humidity": 72,
      "id": 6062055889043456,
      "max_temp": 33.5,
      "min_temp": 24.759999999999998,
      "predictability": 77,
      "the_temp": 31.44,
      "visibility": 9.714517219438479,
      "weather_state_abbr": "hr",
      "weather_state_name": "Heavy Rain",
      "wind_direction": 231.9007449072647,
      "wind_direction_compass": "SW",
      "wind_speed": 4.740631533272356
    },
    {
      "air_pressure": 1009.5,
      "applicable_date": "2020-09-25",
      "created": "2020-09-23T16:00:52.504367Z",
      "humidity": 69,
      "id": 6203842426306560,
      "max_temp": 33.3,
      "min_temp": 25.36,
      "predictability": 75,
      "the_temp": 31.92,
      "visibility": 13.932074186749384,
      "weather_state_abbr": "lr",
      "weather_state_name": "Light Rain",
      "wind_direction": 281.48523379083207,
      "wind_direction_compass": "WNW",
      "wind_speed": 6.146027525416899
    },
    {
      "air_pressure": 1009,
      "applicable_date": "2020-09-26",
      "created": "2020-09-23T16:00:55.494767Z",
      "humidity": 72,
      "id": 6404202113794048,
      "max_temp": 33.125,
      "min_temp": 25.115,
      "predictability": 77,
      "the_temp": 30.545,
      "visibility": 11.966366420106578,
      "weather_state_abbr": "hr",
      "weather_state_name": "Heavy Rain",
      "wind_direction": 286.4991601759091,
      "wind_direction_compass": "WNW",
      "wind_speed": 5.094320465065352
    },
    {
      "air_pressure": 1009,
      "applicable_date": "2020-09-27",
      "created": "2020-09-23T16:00:58.203379Z",
      "humidity": 80,
      "id": 5265468911779840,
      "max_temp": 29.575000000000003,
      "min_temp": 24.18,
      "predictability": 77,
      "the_temp": 28.6,
      "visibility": 9.820461007714945,
      "weather_state_abbr": "hr",
      "weather_state_name": "Heavy Rain",
      "wind_direction": 259.5963365607041,
      "wind_direction_compass": "W",
      "wind_speed": 3.8885642181765157
    },
    {
      "air_pressure": 1009,
      "applicable_date": "2020-09-28",
      "created": "2020-09-23T16:01:02.367098Z",
      "humidity": 77,
      "id": 5425837786529792,
      "max_temp": 30.035,
      "min_temp": 24.165,
      "predictability": 77,
      "the_temp": 29.56,
      "visibility": 9.536183687266364,
      "weather_state_abbr": "hr",
      "weather_state_name": "Heavy Rain",
      "wind_direction": 182.5,
      "wind_direction_compass": "S",
      "wind_speed": 2.9403026326254675
    }
  ],
  "latt_long": "10.759180,106.662498",
  "location_type": "City",
  "parent": {
    "latt_long": "15.974210,107.868042",
    "location_type": "Country",
    "title": "Vietnam",
    "woeid": 23424984
  },
  "sources": [
    {
      "crawl_rate": 360,
      "slug": "bbc",
      "title": "BBC",
      "url": "http://www.bbc.co.uk/weather/"
    },
    {
      "crawl_rate": 480,
      "slug": "forecast-io",
      "title": "Forecast.io",
      "url": "http://forecast.io/"
    },
    {
      "crawl_rate": 180,
      "slug": "met-office",
      "title": "Met Office",
      "url": "http://www.metoffice.gov.uk/"
    },
    {
      "crawl_rate": 360,
      "slug": "openweathermap",
      "title": "OpenWeatherMap",
      "url": "http://openweathermap.org/"
    },
    {
      "crawl_rate": 360,
      "slug": "world-weather-online",
      "title": "World Weather Online",
      "url": "http://www.worldweatheronline.com/"
    }
  ],
  "sun_rise": "2020-09-23T05:42:24.394809+07:00",
  "sun_set": "2020-09-23T17:48:41.332796+07:00",
  "time": "2020-09-23T23:25:11.462420+07:00",
  "timezone": "Asia/Ho_Chi_Minh",
  "timezone_name": "LMT",
  "title": "Ho Chi Minh City",
  "woeid": 1252431
}
