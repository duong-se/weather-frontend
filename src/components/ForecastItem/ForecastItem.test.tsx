import React from 'react'
import { shallow } from 'enzyme'
import ForecastItem from './ForecastItem'
import { mockConsolidatedWeather } from '../../service/__mocks__/mockConsolidatedWeather'

describe('ForecastItem', () => {
  it('should match snapshot', () => {
    const mockOnClick = jest.fn()
    const wrapper = shallow(<ForecastItem consolidatedWeather={mockConsolidatedWeather} onClick={mockOnClick} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should call onClick when click', () => {
    const mockOnClick = jest.fn()
    const wrapper = shallow(<ForecastItem consolidatedWeather={mockConsolidatedWeather} onClick={mockOnClick} />)
    wrapper.simulate('click')
    expect(mockOnClick).toBeCalledTimes(1)
  })
  it('should show correct data', () => {
    const mockOnClick = jest.fn()
    const wrapper = shallow(<ForecastItem consolidatedWeather={mockConsolidatedWeather} onClick={mockOnClick} />)
    const dateText = wrapper.find('.card-subtitle').text()
    const minTempText = wrapper.find('[data-test="min-temp"]').text()
    const maxTempText = wrapper.find('[data-test="max-temp"]').text()
    expect(dateText).toEqual('Tuesday')
    expect(minTempText).toEqual('15℃')
    expect(maxTempText).toEqual('24℃')
  })
})
