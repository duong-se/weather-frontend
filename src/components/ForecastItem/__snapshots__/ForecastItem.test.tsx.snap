// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastItem should match snapshot 1`] = `
<button
  className="m-2 btn btn-light root"
  onClick={[MockFunction]}
>
  <h6
    className="card-subtitle mb-2 mt-2 text-muted textFontSize"
  >
    Tuesday
  </h6>
  <div
    className="weatherStateImg mb-2"
  >
    <img
      alt="lc"
      src="/static/img/weather/lc.svg"
    />
  </div>
  <div
    className="row card-text d-flex temperature"
  >
    <h6
      className="textFontSize"
      data-test="min-temp"
    >
      15
      ℃
    </h6>
    <h6
      className="textFontSize"
      data-test="max-temp"
    >
      24
      ℃
    </h6>
  </div>
</button>
`;
